File: LeetCode-November-challenge/OnlineStockSpan.java
import java.util.*;

class StockSpanner {
    Stack<int[]> ms;

    public StockSpanner() {
        ms = new Stack<>();
    }

    public int next(int price) {
        int res = 1;
        while (!ms.isEmpty() && price >= ms.peek()[0]) {
            res += ms.peek()[1];
            ms.pop();
        }
        ms.add(new int[] { price, res });
        return res;
    }
}

public class OnlineStockSpan {
    public static void main(String[] args) {
        StockSpanner re = new StockSpanner();
        System.out.println(re.next(100));
        System.out.println(re.next(80));
        System.out.println(re.next(60));
        System.out.println(re.next(70));
        System.out.println(re.next(60));
        System.out.println(re.next(75));
        System.out.println(re.next(85));
    }
}

File: LeetCode-November-challenge/OnlineStockSpan.js
var StockSpanner = function () {
    this.stack = [];
};

StockSpanner.prototype.next = function (price) {
    let res = 1;
    const { stack } = this;
    while (stack.length && price >= stack[stack.length - 1][0]) {
        res += stack[stack.length - 1][1];
        stack.pop();
    }
    stack.push([price, res]);
    return res;
};
let re = new StockSpanner();
console.log(re.next(100));
console.log(re.next(80));
console.log(re.next(60));
console.log(re.next(70));
console.log(re.next(60));
console.log(re.next(75));
console.log(re.next(85));